packages/ui/src/components/Stepper/index.tsx

```diff
@@ -21,27 +89,92 @@ const loadingAnimation = keyframes`
   }
 `
 
-const StyledStepContainer = styled.div`
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  justify-content: flex-start;
+const StyledBullet = styled(Bullet)<{
+  size: 'small' | 'medium'
+  isActive: boolean
+}>`
+  margin-top: ${({ theme, size }) =>
+    size === 'small' ? theme.space['0.5'] : 0};
+  transition: all 300ms;
+
+  ${({ theme, isActive }) =>
+    isActive
+      ? `background-color: ${theme.colors.primary.backgroundStrongHover}; 
+      box-shadow: ${theme.shadows.focusPrimary};`
+      : null}
 `
 
 const StyledText = styled(Text)`
   margin-top: ${({ theme }) => theme.space['1']};
+  transition: text-decoration-color 250ms ease-out;
+  text-decoration: underline;
+  text-decoration-thickness: 1px;
+  text-underline-offset: 2px;
+  text-decoration-color: transparent;
+  }
 `
 
-const StyledBullet = styled(Bullet)<{ size: 'small' | 'medium' }>`
-  margin-top: ${({ theme, size }) =>
-    size === 'small' ? theme.space['0.5'] : 0};
+const StyledStepContainer = styled.div<{
+  isActive: boolean
+  isDone: boolean
+  'data-disabled': boolean
+  'data-interactive': boolean
+}>`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: flex-start;
+  transition: all 300ms;
```

Comment:
```suggestion
  transition: all 300ms;
```

This can have side effects with animations, better define exactly what you want to have an animation applied on